src/py/flwr/simulation/ray_transport/ray_actor.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Ray-based Flower Actor and ActorPool implementation."""
+
+import threading
+import traceback
+from logging import ERROR, WARNING
+from typing import Any, Callable, Dict, List, Set
+
+import ray
+from ray.util.actor_pool import ActorPool
+
+from flwr.common.logger import log
+
+
+class ClientException(Exception):
+    """Raised when client side logic crashes with an exception."""
+
+    def __init__(self, message: str):
+        self.message = f"\n{'>'*7} A ClientException occurred." + message
+        super().__init__(self.message)
+
+
+@ray.remote
+class VirtualClientEngineActor:
+    """A Ray Actor class that runs client workloads."""
+
+    def terminate(self):
+        """Terminate Actor."""
+        log(WARNING, f"Manually terminating {self.__class__.__name__}")
+        ray.actor.exit_actor()
+
+    def run(self, client_fn: Callable, client_id):
```

Comment:
`client_fn` might be confused with the `client_fn` that returns a `Client`/`NumPyClient`. I'd propose:
```suggestion
    def run(self, job_fn: Callable, cid: str):
```